### ISSUE TITLE
Issues uploading arduino code

### ISSUE THREAD

**Issue by Cont3mplation** [SammyIAm/Moppy2#16]
Hi all, just started with musical floppy drive projects, and while following the moppy 2.0 wiki startup guide I downloaded the arduino zip and tried to upload the iso, only to get an undefined loop error, and this error seemingly is only solved by adding a void loop(){} and void setup(){} to the code. Is this necessary or is it something else that I'm doing wrong that needs to be fixed? Thanks in advance

**Comment by SammyIAm** [comment]
Did you unzip the Moppy folder from the zip file and open the `.ino` from the unzipped folder?  The `setup()` and `loop()` functions exist in the MoppyCore.cpp file which should be included by the Arduino IDE when compiling.

Another thing worth checking: are you using the latest version of the Arduino IDE?

**Comment by Cont3mplation** [comment]
Oh, the ino I got from the unzipped folders was just comments, am I just being dumb and looking in the wrong folder?

**Comment by Cont3mplation** [comment]
Oh I am just being dumb, found it thanks

**Comment by Cont3mplation** [comment]
Oh okay perhaps not, so I opened the file and got the moppy.ino, and the opened moppyconfig.h in the ide, but neither have the loop or setup. I assume I'm just looking in the wrong place now

**Comment by SammyIAm** [comment]
Alright, just confirmed it works in Arduino IDE 1.8.5 on a fresh computer.  Steps should be:
1. Download `MoppyArduino-2.0.1.zip`
1. Extract the contents of the zip file (the results should be a folder called "Moppy")
1. Open Moppy.ino from inside the *unzipped* folder.

The IDE should show three tabs: "Moppy", "MoppyConfig.h", and "MoppyCore.cpp".  You may want to open a new sketch in the IDE and make sure the existing one is closed and then try again (maybe the IDE is confused because you've opened two sketches called Moppy?)

**Comment by Cont3mplation** [comment]
Ah okay, mine didn't open the c++ file, that might be part of the issue will check :D Thanks for taking the time to help btw man, much appreciated!

**Comment by Cont3mplation** [comment]
Okay having retried that it worked fine, thanks for the help! But then when I went to run the .bat file in the bin (after having extracted it) it just opened a blank commandline then closed, and nothing happened. Same problem occurred when I ran it with admin permissions, and having checked the wiki it doesn't seem to have anything with issues like this. Do I need to have any external programs installed for it to work correctly or am I simply doing something wrong? (appreciate your patience by the way man, sorry for just spamming you with issues)
